File: CS/Reporting-How-To-Use-Object-Data-Source/WebForms/VB/Reporting_ObjectDS_WebForms/Services/ObjectDataSourceWizardCustomTypeProvider.vb
Imports DevExpress.DataAccess.Web
Imports System
Imports System.Collections.Generic

Namespace Reporting_ObjectDS_WebForms.Services
	Public Class ObjectDataSourceWizardCustomTypeProvider
		Implements IObjectDataSourceWizardTypeProvider

		Public Function GetAvailableTypes(ByVal context As String) As IEnumerable(Of Type) Implements IObjectDataSourceWizardTypeProvider.GetAvailableTypes
			Return {GetType(Employees.EmployeeList)}
		End Function
	End Class
End Namespace

File: WebForms/CS/Reporting_ObjectDS_WebForms/Services/ObjectDataSourceWizardCustomTypeProvider.cs
using DevExpress.DataAccess.Web;
using System;
using System.Collections.Generic;

namespace Reporting_ObjectDS_WebForms.Services {
    public class ObjectDataSourceWizardCustomTypeProvider : IObjectDataSourceWizardTypeProvider {
        public IEnumerable<Type> GetAvailableTypes(string context)
        {
            return new[] { typeof(Employees.EmployeeList) };
        }
    }
}
